### DISCUSSION TITLE
Implement a API recursion warning or perhaps when debug is True a print()/log of functions being added to js_api.

### DISCUSSION THREAD

**Discussion by devdave** [r0x0r/pywebview#1326] [Ideas]
Just upgraded last night from 4.41 to the latest 5.x.   

For my project I have something  like

```python
class MyAPI:
     app:Application
     db_path:pathlib.Path

     def some_func(self,...)
          ...

```

With the newish recursive behavior, it exposed not just `MyAPI` but also the `MyAPI.app` and then got stuck on a infinite recursion with `db__path.parent().parent()...`.    I was able to fix the problem by hacking in `print` statements into the js api util.py. figure out what was happening, and then prefixing the API class's properties with `__`.   A useful feature request might be something like those having conditional print statements in the js api expose/import process.  

Otherwise, I am looking forward to figuring out how to exploit the `{basename}.{name}` functionality to clean/tidy my project up.

**Comment by r0x0r** [comment]
I have committed an infinite recursion prevention check to master

**Reply by devdave** [reply]
Wow, thank you.
